### ISSUE TITLE
[doc] Remove README.md from docs.opentitan.org

### ISSUE THREAD

**Issue by moidx** [lowRISC/opentitan#13345]
README.md files are often created to provide in source reference documentation, and not always contain valid Hugo Markdown metadata.  This makes the documentation autogenerated indexes and the overall website structure hard to follow. 

This issue proposes the following changes: 

- [x] Move all docs.opentitan.org relevant documentation to `{_}index.md` and or `docs/.md` files. 
- [x] Update all website documentation to have proper Hugo metadata.  
- [x] Remove README.md file from docs.opentitan.org.

This new documentation folder structure will make the organization more intuitive, making it more explicit what we want to publish on the documentation website, while also allowing developers to use README.md files for in source documentation.

**Comment by drewmacrae** [comment]
SGTM, README.md is a sufficiently typical and useful name that shouldn't confuse anyone and there's going to be a version hosted on Github that could be pointed to by a URL if it's important to make reference to it.

**Comment by msfschaffner** [comment]
Related: #13345

**Comment by arunthomas** [comment]
SGTM too

**Comment by marnovandermaas** [comment]
There's quite a few README pages on docs.opentitan.org. With this plan I'm wondering about the following:
- Does this double the documentation workload in certain places?
- How do we avoid documentation fragmentation where some info is in the README and some in the docs?
- Is there an easy way to update and enforce that OT READMEs have proper Hugo metadata (e.g. linting)?

**Comment by moidx** [comment]
> There's quite a few README pages on docs.opentitan.org. With this plan I'm wondering about the following:
> 
> * Does this double the documentation workload in certain places?

There is no need to double the workload. Adding README files to the documentation website was not intended as a feature. Using `doc/...` and `{_}index.md` directives allows us to be more explicit about the content we post on the website, and also allow us to make README files more ergonomic for in repo rendering (e.g. GitHub, code search).

> * How do we avoid documentation fragmentation where some info is in the README and some in the docs?

We'll provide some guidelines on what is recommended to go where. If the documentation needs to point to a README file it can still do so, but the reference will point to Github. 

> * Is there an easy way to update and enforce that OT READMEs have proper Hugo metadata (e.g. linting)?

This is possible, but given the state of the website, I rather make a clear distinction between README files and documentation intended to go on the website.

**Comment by moidx** [comment]
The OTBN README is a great example of how I think we should be using README files: https://github.com/lowRISC/opentitan/tree/master/hw/ip/otbn

It has pointers to relevant documentation, and provides a quick reference on how to get started. It also renders well in both GitHub and [code search](https://cs.opensource.google/opentitan/opentitan/+/master:hw/ip/otbn/) given that it does not use any Hugo markup.

**Comment by moidx** [comment]
Closing as implemented.
